src/Survey/ItechArt.Survey.Foundation/Authentication/Validation/UserValidator.cs

```diff
@@ -0,0 +1,119 @@
+﻿using ItechArt.Common.Validation;
+using ItechArt.Common.Validation.Abstractions;
+using ItechArt.Survey.DomainModel;
+using ItechArt.Survey.Foundation.Authentication.Abstractions;
+using ItechArt.Survey.Foundation.Authentication.Configuration;
+using ItechArt.Survey.Foundation.Authentication.Validation.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace ItechArt.Survey.Foundation.Authentication.Validation;
+
+public class UserValidator : Validator<User, UserRegistrationErrors>, IUserValidator
+{
+    private readonly RegistrationOptions _options;
+
+
+    public UserValidator(IOptions<RegistrationOptions> options)
+    {
+        _options = options.Value;
+    }
+
+
+    public ValidationResult<UserRegistrationErrors> ValidatePassword(string password)
+    {
+        var error = ValidateUserPassword(password);
+
+        if (error.HasValue)
+        {
+            return ValidationResult<UserRegistrationErrors>.CreateUnsuccessful(error.Value);
+        }
+
+        return ValidationResult<UserRegistrationErrors>.CreateSuccessful();
+    }
+
+
+    protected override UserRegistrationErrors? ValidateWithErrorReturning(User user)
+    {
+        var error = ValidateUserName(user.UserName);
+
+        if (error != null)
+        {
+            return error;
+        }
+
+        error = ValidateUserEmail(user.Email);
```

Comment:
create a separate variable, don't reuse variable for different purposes (apply everywhere)